### DISCUSSION TITLE
[Resolved] get_comparable_java_version() not working as expected

### DISCUSSION THREAD

**Discussion by jensppi** [tofi86/universalJavaApplicationStub#118] [General]
Hallo,

first of all thanx a lot for this script.

While I was manipulating the script for my needs I noticed the following inconsistency:

The comment of get_comparable_java_version() says

`@return  an 8 digit numeral ('1.8.0_45'->'08000045'; '9.1.13'->'09001013')`

but instead the return value is

`' 8  0 45' or ' 9  1 13'`

The three parts of the version number are left padded by spaces not by zeros. 

The three printf in the function get_comparable_java_version() are using %03**s** not %03**d** as expected.

This is not a bug, cause your script seems to work since years with this inconsistency. It's only a hint.

Regards,
Jens

**Comment by jensppi** [comment]
I apologize. get_comparable_java_version() is working as expected in a macOS bash. I used a Linux bash to test some code. In a linux bash 
echo "$(printf '%02s' 4)"
is
 4
On a macOS bash it is 
04

Sorry: Don't know how to delete a discussion.

**Comment by tofi86** [comment]
Hi @jensppi, nevermind. I think we cannot close a discussion, so we'll just have to leave this as is. I'll change the subject though...
Best, Tobias
